guid_renderer.py

```diff
@@ -0,0 +1,30 @@
+from binaryninja import DataRenderer, DisassemblyTextLine, InstructionTextToken, InstructionTextTokenType, BinaryReader, \
+    log_alert, BinaryView, VoidType, HighLevelILCallSsa
+import binaryninja
+
+
+class EfiGuidDataRenderer(DataRenderer):
+    def __init__(self):
+        DataRenderer.__init__(self)
+
+    def perform_is_valid_for_data(self, ctxt, view, addr, typ, context):
+        return DataRenderer.is_type_of_struct_name(typ, "EFI_GUID", context)
+
+    def perform_get_lines_for_data(self, ctxt, view, addr, typ, prefix, width, context):
+        result = [DisassemblyTextLine(prefix, addr)]
```

Comment:
Recommend simplifying with something like

```python
from uuid import UUID
guid_str = str(UUID(bytes=bv.read(16)))
```

Reply:
The result of `UUID` has a different order from `EFI_GUID`, and the result of `UUID` has a different bit order. 

```
struct EFI_GUID EFI_LOADED_IMAGE_PROTOCOL_GUID = 
    [EFI_GUID("5b1b31a1-9562-11d2-8e3f00a0c969723b")]
```

```
>>> str(UUID(bytes=bv.read(addr, 16)))
'a1311b5b-6295-d211-8e3f-00a0c969723b'
```

This class is simply modified from this [post](https://binary.ninja/2024/04/08/customizing-data-display.html), and I don't know if there is a better way to do that.